pkg/handler/employee/employee.go

```diff
@@ -290,3 +291,102 @@ func (h *handler) UpdateGeneralInfo(c *gin.Context) {
 	}
 	c.JSON(http.StatusOK, view.CreateResponse[any](view.ToEmployeeData(employee), nil, nil, nil))
 }
+
+// Create godoc
+// @Summary Create new employee
+// @Description Create new employee
+// @Tags Employee
+// @Accept  json
+// @Produce  json
+// @Success 200 {object} view.EmployeeData
+// @Failure 400 {object} view.ErrorResponse
+// @Failure 404 {object} view.ErrorResponse
+// @Failure 500 {object} view.ErrorResponse
+// @Router /employee [post]
+func (h *handler) Create(c *gin.Context) {
+	// 1. parse employee data from body
+	var req CreateEmployee
+
+	if err := c.ShouldBindJSON(&req); err != nil {
+		c.JSON(http.StatusBadRequest, view.CreateResponse[any](nil, nil, err, req))
+		return
+	}
+
+	// 1.1 prepare the logger
+	l := h.logger.Fields(logger.Fields{
+		"handler": "employee",
+		"method":  "One",
+		"params":  req,
+	})
+
+	// 1.2 prepare employee data
+	now := time.Now()
+	pos, err := h.store.Position.One(req.PositionID)
+	if err != nil {
+		l.Error(err, "error invalid position")
+		if err == gorm.ErrRecordNotFound {
+			c.JSON(http.StatusBadRequest, view.CreateResponse[any](nil, nil, ErrPositionNotfound, req))
+			return
+		}
+		c.JSON(http.StatusInternalServerError, view.CreateResponse[any](nil, nil, err, req))
+		return
+	}
+	sen, err := h.store.Seniority.One(req.SeniorityID)
+	if err != nil {
+		l.Error(err, "error invalid seniority")
+		if err == gorm.ErrRecordNotFound {
+			c.JSON(http.StatusBadRequest, view.CreateResponse[any](nil, nil, ErrSeniorityNotfound, req))
+			return
+		}
+		c.JSON(http.StatusInternalServerError, view.CreateResponse[any](nil, nil, err, req))
+		return
+	}
+	role, err := h.store.Role.One(req.RoleID)
+	if err != nil {
+		l.Error(err, "error invalid role")
+		if err == gorm.ErrRecordNotFound {
+			c.JSON(http.StatusBadRequest, view.CreateResponse[any](nil, nil, ErrRoleNotfound, req))
+			return
+		}
+		c.JSON(http.StatusInternalServerError, view.CreateResponse[any](nil, nil, err, req))
+		return
+	}
+	employee := &model.Employee{
+		BaseModel: model.BaseModel{
+			ID: model.NewUUID(),
+		},
+		FullName:      req.FullName,
+		DisplayName:   req.DisplayName,
+		TeamEmail:     req.TeamEmail,
+		PersonalEmail: req.PersonalEmail,
+		WorkingStatus: model.WorkingStatusProbation,
+		JoinedDate:    &now,
```

Comment:
I'm not sure JoneDate is the create record time or the time employee become Dwarves's member ? cc anh @namnhce

Reply:
for now we use current time to determine joined date.